Modules/posixmodule.c

```diff
@@ -613,11 +613,16 @@ PyOS_BeforeFork(void)
     run_at_forkers(interp->before_forkers, 1);
 
     _PyImport_AcquireLock(interp);
+    _PyEval_StopTheWorldAll(&_PyRuntime);
```

Comment:
Consider adding a comment to the effect of "The child won't have to start the world again because the current thread will be the only one left."

Reply:
We start the world again in the `_PyThreadState_DeleteExcept` call, which is in turn called by `_PyEval_ReInitThreads`. Even though there is only one thread left, we still want to undo the setup performed by `_PyEval_StopTheWorldAll` (like unlocking the stop-the-world mutex).

The placement in `_PyThreadState_DeleteExcept` may seem a bit strange, but it's required for the shutdown use case: we need to start the world again *after* we remove thread states from the interpreter linked list, but before we call `PyThreadState_Clear()`, which may call arbitrary Python code.

Reply:
I think this would be a bit cleaner if we split up `_PyThreadState_DeleteExcept` into two functions:

1) One that removes thread states from the linked list
2) One that calls `PyThreadState_Clear()` on the removed thread states

And have the caller explicitly call `_PyEval_StartTheWorldAll(runtime);` in between the two functions.

However, I went with the smaller change in this PR

Reply:
that refactoring makes sense to me.  I'll leave it up to you as to whether you want to do that in a separate PR.

Reply:
> We start the world again in the _PyThreadState_DeleteExcept call

Ah, I missed that.  Thanks for clarifying.

Reply:
> I'll leave it up to you as to whether you want to do that in a separate PR.

I'll do it in a follow up PR.

Reply:
Follow up PR: https://github.com/python/cpython/pull/117131